zerver/models.py

```diff
@@ -2144,6 +2144,18 @@ def get_user_by_delivery_email(email: str, realm: Realm) -> UserProfile:
     return UserProfile.objects.select_related().get(
         delivery_email__iexact=email.strip(), realm=realm)
 
+def get_users_by_delivery_email(emails: Set[str], realm: Realm) -> QuerySet:
+    """This is similar to get_users_by_delivery_email, and
+    it has the same security caveats.  It gets multiple
+    users and returns a QuerySet, since most callers
+    only will grab two or three fields.
+    """
+    email_filter = Q()
+    for email in emails:
+        email_filter |= Q(delivery_email__iexact=email.strip())
```

Comment:
Can we just use `delivery_email__iexact__in=` and pass a list, rather than using `Q`?  That might be more readable and also produce between SQL.

Reply:
It's a Django limitation.  Try this in the shell:

~~~
In [2]: list(UserProfile.objects.filter(email__iexact='x'))
In [3]: list(UserProfile.objects.filter(email__in=['x']))
In [4]: list(UserProfile.objects.filter(email__iexact__in=['x']))
~~~

The first two work, the third explodes.

See https://stackoverflow.com/questions/14907525/how-can-i-chain-djangos-in-and-iexact-queryset-field-lookups.

Reply:
Good to know :)